### DISCUSSION TITLE
Does function pointers require strong GC reference?

### DISCUSSION THREAD

**Discussion by OwnageIsMagic** [dotnet/runtime#91896] [Q&A]
If `extern` method captures function pointer (`delegate* {managed or unmanaged}`) do I need to keep strong reference to it on managed side?
Is function jited when it's address taken (`&`)? If not, it seems like I need to later update stored pointer, so it will point to actual implementation instead of stub (for better performance).
How does it play with tiered jit (Tier0 is never deallocated?)? What about assembly unloading?

**Comment by alexrp** [comment]
> If `extern` method captures function pointer (`delegate* {managed or unmanaged}`) do I need to keep strong reference to it on managed side?

The function pointer can be invalidated if the assembly is unloaded (see `AssemblyLoadContext`). But other than that, no; it's not like `Marshal.GetFunctionPointerForDelegate()` where you have to keep the delegate alive manually.
